### PR TITLE
[WIP] Change ArticleForm layout to improve UX on Android

### PR DESCRIPTION
**Pull request by tudorpavel**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description

This PR still needs tidying up, hence the WIP. I just want to confirm if this is a good approach or not.

I want to help with this but of course there isn't an easy fix.

There are issues on Android browsers when using dynamic element heights and/or fixed positioning while opening the keyboard. The reason is that when opening the keyboard on Android browsers, the body height will change by subtracting the keyboard height. Dynamic heights will change while the keyboard is opening and cause the browser's "autoscroll to cursor" behavior to go nuts.

I managed to make the behavior a bit better by changing the layout:
  - I removed `preact-textarea-autosize`, added a hardcoded `height: 300px` to the textarea and re-introduced the scroll
  - bottom buttons are no longer with `position: fixed`

I'm not sure what else needs to change, for example I noticed that sometimes there's a `v1` for `articleform__form articleform__form--v2` about which I'm not sure but we can discuss it here.

## Related Tickets & Documents

Issue #4385 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Result in Firefox:
firefox

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

it's fine

or

just css things

### CODE REVIEW

**Comment by tudorpavel** [comment]
@jessleenyc @rhymes let me know what you think about this approach. If it's too big of a change, we can close this PR before it becomes stale.

**Comment by rhymes** [comment]
@tudorpavel FYI, WIP or drafts PRs usually mean "don't review it yet, I'm finalizing it", that's probably why no one has reviewed it so far. 

I wonder if the hardcoded textarea size doesn't break anything with other devices or formfactors.

ps. there's a failing test :)

**Comment by tudorpavel** [comment]
> @tudorpavel FYI, WIP or drafts PRs usually mean "don't review it yet, I'm finalizing it", that's probably why no one has reviewed it so far.
> 
> I wonder if the hardcoded textarea size doesn't break anything with other devices or formfactors.
> 
> ps. there's a failing test :)

Oh okay, I wasn't sure how to open a PR that says "here's a rough idea we could try, let's discuss it". :smile: That's why it's not polished and since it's a refactor of the UI layout I wanted to ask someone's opinion before moving along down this path too much, you know?

Indeed, even with the hardcoded textarea height, some of the same issues still happen in landscape mode because of the small screen height.

I took inspiration from Twitter's web app for mobile, whose "Compose Tweet" form also has a limited height with scrollable textarea. Their app is also experiencing these same issues in landscape because of nested scrolls (scrollable textarea inside scrollable page). When focusing the cursor inside a scrollable textarea, the browser opens the keyboard and then tries to scroll to bring the cursor into view, but it seems to only scroll the textarea and not the page.

So long story short, I'm not yet sure what would be an ideal solution for the ArticleForm.

**Comment by mstruve** [comment]
Hey @tudorpavel! 

Thank you so much for this PR! Currently, we are in the process of going through and cleaning up some of our PR history to help us better prioritize what we need to be focusing on. Since this PR has not been active for a while and has some conflicts I am going to close it for now. 

If you would still like to get this merged please feel free to resolve the conflicts and then open the PR back up. We would be more than happy to help push this through if you would like! Also when you open the PR up make sure it is ready to be reviewed and does not have the WIP tag. That will indicate to our team to review it. We usually don't look at WIP PRs bc it means they are not ready for review. 

If you would like team help on it we suggest opening a Draft PR and then leaving a comment that you need help and we will do our best to find some time to assist you. 

Thanks so much for being a contributor! Ping us anytime if you have questions!
